torchvision/transforms/v2/_auto_augment.py

```diff
@@ -621,3 +696,269 @@ def forward(self, *inputs: Any) -> Any:
+                f"but found {len(bboxes_list)}."
+            )
+
+        idx1, image_or_video = image_or_videos[0]
+        idx2, bboxes = bboxes_list[0]
+        return (flat_inputs, spec, idx1, idx2), image_or_video, bboxes
+
+    def _unflatten_and_insert_image_or_video_and_bboxes(
+        self,
+        flat_inputs_with_spec: Tuple[List[Any], TreeSpec, int, int],
+        image_or_video: Union[datapoints._ImageType, datapoints._VideoType],
+        bboxes: datapoints.BoundingBox,
+    ) -> Any:
+        flat_inputs, spec, idx1, idx2 = flat_inputs_with_spec
+        flat_inputs[idx1] = image_or_video
+        flat_inputs[idx2] = bboxes
+        return tree_unflatten(flat_inputs, spec)
+
+    @staticmethod
+    def _transform_image_or_video_in_bboxes(
+        fn: Callable[..., torch.Tensor], fn_kwrgs: dict, image: torch.Tensor, bboxes: torch.Tensor
+    ) -> torch.Tensor:
+        # TODO: ? Do we need to assert bboxes format is XYXY?
+        new_img = image.clone()
+        int_bboxes = bboxes.to(torch.long)
+        for bbox in int_bboxes:
+            bbox_img = new_img[..., bbox[1] : bbox[3], bbox[0] : bbox[2]]
+            out_bbox_img = fn(bbox_img, **fn_kwrgs)
+            new_img[..., bbox[1] : bbox[3], bbox[0] : bbox[2]] = out_bbox_img
+        return new_img
+
+    def _apply_image_or_video_and_bboxes_transform(
+        self,
+        image: Union[datapoints._ImageType, datapoints._VideoType],
+        bboxes: datapoints.BoundingBox,
+        transform_id: str,
+        magnitude: float,
+        interpolation: InterpolationMode,
+        fill: Dict[Type, datapoints._FillTypeJIT],
+    ) -> Tuple[Any, datapoints.BoundingBox]:
+        if transform_id == "BBox_Cutout":
+            random_index = int(torch.randint(len(bboxes), ()))
+            # TODO: this wrap_like solution seems very complex
+            chosen_bbox = bboxes.wrap_like(bboxes, bboxes[random_index].unsqueeze(0))
+            fn_kwargs = dict(transform_id=transform_id, magnitude=magnitude, interpolation=interpolation, fill=fill)
+            image = self._transform_image_or_video_in_bboxes(
+                self._apply_image_or_video_transform, fn_kwargs, image, chosen_bbox
+            )
+        elif transform_id.endswith("_Only_BBoxes"):
+            transform_id = transform_id.replace("_Only_BBoxes", "")
+            fn_kwargs = dict(transform_id=transform_id, magnitude=magnitude, interpolation=interpolation, fill=fill)
+            image = self._transform_image_or_video_in_bboxes(
+                self._apply_image_or_video_transform, fn_kwargs, image, bboxes
+            )
+        else:
+            image = self._apply_transform(image, transform_id, magnitude, interpolation, fill)
+            bboxes = self._apply_transform(bboxes, transform_id, magnitude, interpolation, fill)
+
+        return image, bboxes
+
+
+class AutoAugmentDetection(_AutoAugmentDetectionBase):
+    _AUGMENTATION_SPACE = {
```

Comment:
I'm now exactly following [tf repo setting](https://github.com/tensorflow/tpu/blob/b24729de804fdb751b06467d3dce0637fa652060/models/official/efficientnet/autoaugment.py) to set the level ranges, which means level ranges from [0, 10] (0 and 10 included).